src/support/lock_ext.c

```diff
@@ -0,0 +1,74 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_ext_spin_init --
+ *     Allocate and initialize a spinlock.
+ */
+int
+__wt_ext_spin_init(WT_EXTENSION_API *wt_api, WT_SESSION *session, void *spin_retp, const char *name)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *default_session;
+    WT_SPINLOCK *lock;
+
+    (void)session; /*Unused parameters */
+    *(void **)spin_retp = NULL;
+    default_session = ((WT_CONNECTION_IMPL *)wt_api->conn)->default_session;
+    if ((ret = __wt_malloc(default_session, sizeof(WT_SPINLOCK), &lock)) != 0)
```

Comment:
I'd prefer to use calloc_one - I *think* we prefer initialized memory allocations except for performance paths.